core/whistle_apps-1.0.0/src/whapps_maintenance.erl

```diff
@@ -949,3 +951,20 @@ show_status(CallId, 'true', Resp) ->
     lager:info("Media Server: ~s", [wh_json:get_value(<<"Switch-Hostname">>, Resp)]),
     lager:info("Responding App: ~s", [wh_json:get_value(<<"App-Name">>, Resp)]),
     lager:info("Responding Node: ~s", [wh_json:get_value(<<"Node">>, Resp)]).
+
+
+-spec delete_system_media_references() -> 'no_return'.
+delete_system_media_references() ->
+    DocId = <<"media">>,
+    {'ok', MediaDoc} = couch_mgr:open_cache_doc(?WH_CONFIG_DB, DocId),
+    TheKey = <<"default">>,
+    MediaDefault = wh_json:get_value(TheKey, MediaDoc),
+    Deleter =
+        fun (_Key, <<"/system_media/",_/binary>>, Acc) -> Acc;
+            (Key, Value, Acc) -> [{Key, Value} | Acc]
+        end,
+    NewMediaDefault = wh_json:foldl(Deleter, [], MediaDefault),
```

Comment:
This should be `wh_json:new()` instead of `[]`. `wh_json:set_value/3` should be setting `TheKey` to a JSON object, not a list of Key/Value pairs. `Deleter` should use `wh_json:set_value/3` to set them in the Acc too.

Finally, the code should detect if any changes were made and only save if so.